neuraxle/metaopt/auto_ml.py

```diff
@@ -641,60 +641,35 @@ def _fit_data_container(self, data_container: DataContainer, context: ExecutionC
-                                trial_number + 1,
-                                self.n_trial,
-                                repo_trial_split.split_number + 1,
-                                len(training_data_container),
-                                json.dumps(repo_trial.hyperparams, sort_keys=True, indent=4),
-                                # TODO: make hyperparams space serializable
-                                # json.dumps(auto_ml_data.hyperparameter_space.to_flat_as_dict_primitive(), sort_keys=True, indent=4),
-                            )
-                            self.print_func('fitting trial {}'.format(
-                                trial_split_description
-                            ))
-
-                            repo_trial_split = self.trainer.fit_trial_split(
-                                trial_split=repo_trial_split,
-                                train_data_container=training_data_container,
-                                validation_data_container=validation_data_container,
-                                context=context
-                            )
-
-                            repo_trial_split.set_success()
-
-                            self.print_func('success trial {} score: {}'.format(
-                                trial_split_description,
-                                repo_trial_split.get_validation_score()
-                            ))
-
-                        except Exception as error:
-                            track = traceback.format_exc()
-                            self.print_func('failed trial {}'.format(
-                                trial_split_description
-                            ))
-                            self.print_func(track)
-                            repo_trial_split.set_failed(error)
-
-                    repo_trial.update_final_trial_status()
-
-            self.hyperparams_repository.save_trial(repo_trial)
+            try:
+                auto_ml_data = AutoMLContainer(
+                    trial_number=trial_number,
+                    trials=self.hyperparams_repository.load_all_trials(TRIAL_STATUS.SUCCESS),
+                    hyperparameter_space=self.pipeline.get_hyperparams_space(),
+                    main_scoring_metric_name=self.trainer.get_main_metric_name()
+                )
+
+                with self.hyperparams_repository.new_trial(auto_ml_data) as repo_trial:
+                    self.print_func('\ntrial {}/{}'.format(trial_number + 1, self.n_trial))
+
+                    repo_trial_split = self._execute_trial(
+                        trial_number=trial_number,
+                        repo_trial=repo_trial,
+                        context=context,
+                        validation_splits=validation_splits
+                    )
+            except (SystemError, SystemExit, EOFError, KeyboardInterrupt) as error:
+                track = traceback.format_exc()
+                repo_trial.set_failed(error)
```

Comment:
We could probably have a value for cancelled trials, but it's ok for now.

Reply:
https://github.com/Neuraxio/Neuraxle/issues/305